tests/check_ipc.c

```diff
@@ -265,10 +265,12 @@ s1_msg_process_fn(qb_ipcs_connection_t *c,
 		}
 
 	} else if (req_pt->id == IPC_MSG_REQ_SERVER_FAIL) {
+		qb_ipcs_destroy(s1);
 		exit(0);
 	} else if (req_pt->id == IPC_MSG_REQ_SERVER_DISCONNECT) {
 		multiple_connections = QB_FALSE;
 		qb_ipcs_disconnect(c);
+		qb_ipcs_destroy(s1);
```

Comment:
Overall, the code logic is quite hard to follow, so I would encourage adding
comments along whenever there's something not entirely apparent.

In this very case, I have no clue why you wouldn't want to substitute
`disconnect` for `destroy`, but you put it in addition instead, why a superficial
look at the latter suggests the former is implied...